language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala

```diff
@@ -248,8 +248,31 @@ class DottyLanguageServer extends LanguageServer
     implicit val ctx = driver.currentCtx
 
     val pos = sourcePosition(driver, uri, params.getPosition)
-    val enclTree = Interactive.enclosingTree(driver.openedTrees(uri), pos)
-    val sym = Interactive.sourceSymbol(enclTree.symbol)
+    val path = Interactive.pathTo(driver.openedTrees(uri), pos)
+    val enclTree = Interactive.enclosingTree(path)
+
+    val sym = {
+      val sym = path match {
+        // For a named arg, find the target `DefDef` and jump to the param
+        case NamedArg(name, _) :: Apply(fn, _) :: _ =>
+          val funSym = fn.symbol
+          if (funSym.name == StdNames.nme.copy
+            && funSym.is(Synthetic)
+            && funSym.owner.is(CaseClass)) {
+            funSym.owner.info.member(name).symbol
+          } else {
+            val classTree = funSym.topLevelClass.asClass.tree
+            tpd.defPath(funSym, classTree).lastOption.flatMap {
```

Comment:
This expression is a bit dense. I believe the following would be slightly longer, but more readable:
```scala
val paramSymbol = 
  for {
    DefDef(_, _, paramss, _, _) <- tpd.defPath(funSym, classTree).lastOption
    param <- paramss.flatten.find(_.name == name)
  } yield param.symbol
paramSymbol.getOrElse(fn.symbol)
```